cabal-install/src/Distribution/Client/Init/Types.hs

```diff
@@ -282,13 +289,23 @@ mkLiterate _ hs = hs
 -- -------------------------------------------------------------------- --
 -- Interactive prompt monad
 
+type PromptIO = ReaderT (Data.IORef.IORef SessionState) IO
+
+runPromptIO :: PromptIO a -> IO a
+runPromptIO pio =
+  (Data.IORef.newIORef _newSessionState) >>= (runReaderT pio)
+
 newtype PurePrompt a = PurePrompt
-  { _runPrompt
-      :: NonEmpty String
-      -> Either BreakException (a, NonEmpty String)
+  { _runPromptState
+      :: (NonEmpty String, SessionState)
+      -> Either BreakException (a, (NonEmpty String, SessionState))
   }
   deriving (Functor)
 
+_runPrompt :: PurePrompt a -> NonEmpty String -> Either BreakException (a, NonEmpty String)
```

Comment:
I don't understand why underscores were used in this file before, and I don't think it makes sense to increase their usage. Do you have a rule you follow when choosing to use the underscore?

Reply:
The usual rule is to use a prefix underscore for something that might not be used. It's generally a bad idea to deviate from it, since people will make assumptions according to that rule.

Reply:
Agreed!

Reply:
I noticed that before this PR _runPrompt was defined in the record was with a _ so I kept it. I though it was signaling functions that are internal to the module or should be but for some reason they aren't. I dropped a bunch of _.

Reply:
Good!